pandas/core/frame.py

```diff
@@ -3578,7 +3579,14 @@ def extract_unique_dtypes_from_dtypes_set(
                 unique_dtype
                 for unique_dtype in unique_dtypes
                 if issubclass(unique_dtype.type, tuple(dtypes_set))  # type: ignore
+                or (
+                    np.number in dtypes_set
+                    and is_extension_array_dtype(unique_dtype)
+                    and unique_dtype._is_numeric
+                )
             ]
+            if np.number in dtypes_set:
+                extracted_dtypes.extend([])
```

Comment:
not looked at this in detail, but from the name of the function, i'm assuming that this is not the correct location for this special casing.

There is another issue regarding extension arrays and dtypes, https://github.com/pandas-dev/pandas/issues/9581. maybe a compat function for np.issubdtype could help, where we could add the special cases for EAs.

Reply:
OK I've added function to pandas\compat\numpy\__init__.py